app/forms/register_user_email_form.rb

```diff
@@ -35,11 +35,22 @@ def valid_form?
   def process_errors
     # To prevent discovery of existing emails, we check to see if the email is
     # already taken and if so, we act as if the user registration was successful.
-    if email_taken?
+    if email_taken? && user_unconfirmed?
```

Comment:
Maybe for another time, but I was trying to trace down where this `email_taken?` comes from...and finding out the `@email_taken` var is as a side effect of `email_is_unique` was kinda surprising.

Also we end up doing 2 queries to the users table by emails for this? (one for `.exists?` in `email_is_unique` and another in `existing_user`)